pool.go

```diff
@@ -0,0 +1,63 @@
+package Go_Pool
+
+import (
+	"errors"
+	"runtime"
+	"sync"
+)
+
+type Pool struct {
+	Tasks       []*Task
+	concurrency int
+	numCPUs     int
+	tasksChan   chan *Task
+	wg          sync.WaitGroup
+}
+
+var pool *Pool
+var once sync.Once
+
+func NewPool(tasks []*Task, concurrency int, numCPUs int) *Pool {
+	once.Do(func() {
+		pool = &Pool{
+			Tasks:       tasks,
+			concurrency: concurrency,
+			tasksChan:   make(chan *Task, concurrency),
+		}
+	})
+	pool.SetNumCPUs(numCPUs)
```

Comment:
It would be better to use this option only while starting the pool server.

Reply:
Yeah but doing like that we won't be able to update the no of CPUs when creating new objects of it as it is a singleton object.